packages/providers/src/utils/sleep.ts

```diff
@@ -0,0 +1,7 @@
+export function sleep(time: number = 1000) {
```

Comment:
I'd discourage using a default value for this parameter since we can't assume anything about it.

> **Note** _We could replace this with [`sleep-promise`](https://www.npmjs.com/package/sleep-promise), but perhaps that'd be trickier to mock in tests, in which case having a simple function may still have its place._

```suggestion
export function sleep(time: number) {
```